onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -2918,6 +2979,76 @@ Status ResizeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const
 
 #pragma endregion
 
+#pragma region op_flatten
+
+class FlattenOpBuilder : public BaseOpBuilder {
+ private:
+  bool IsOpSupportedImpl(ModelBuilder& /* model_builder */, const Node& node) override;
+  static void GetFlattenShape(const Node& node, const Shape& input_shape, int32_t& dim_1, int32_t& dim_2);
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node) override ORT_MUST_USE_RESULT;
+};
+
+/* static */ void FlattenOpBuilder::GetFlattenShape(const Node& node, const Shape& input_shape, int32_t& dim_1, int32_t& dim_2) {
+  int32_t rank = SafeInt<int>(input_shape.size());
+  NodeAttrHelper helper(node);
```

Comment:
SafeInt is more for making sure a memory allocation is safe, so doesn't need to be used everywhere.

I don't think it's adding value here either as you're converting straight back to int32_t and not doing any calculations with the value that could overflow.

Reply:
Changed to static_cast